tests/system_tests_two_routers.py

```diff
@@ -470,19 +497,17 @@ def on_message(self, event):
                                 'adminStatus': 'deleted'}
                             request.reply_to = self.mgmt_receiver_1.remote_source.address
                             self.mgmt_sender.send(request)
+                            self.n_sent += 1
```

Comment:
If you add the additional check to wait for "receiver_to_kill", then the test should immediately fail if this loop exits without finding the connection to delete.